lib/core/presentation/pages/quest/quest_listing_page/quest_listing_page.dart

```diff
@@ -0,0 +1,203 @@
+import 'package:app/theme/sizing.dart';
+import 'package:app/theme/spacing.dart';
+import 'package:app/theme/typo.dart';
+import 'package:auto_route/auto_route.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+
+enum QuestListingTabs {
+  profile(tabIndex: 0),
+  post(tabIndex: 1),
+  event(tabIndex: 2),
+  citizenship(tabIndex: 3);
+
+  const QuestListingTabs({
+    required this.tabIndex,
+  });
+
+  final int tabIndex;
+}
+
+@RoutePage()
+class QuestListingPage extends StatefulWidget {
+  const QuestListingPage({
+    super.key,
+  });
+
+  @override
+  State<QuestListingPage> createState() => _QuestListingPageState();
+}
+
+class _QuestListingPageState extends State<QuestListingPage>
+    with SingleTickerProviderStateMixin {
+  late int selectedTabIndex = QuestListingTabs.profile.index;
+
+  @override
+  void initState() {
+    super.initState();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final t = Translations.of(context);
+    final colorScheme = Theme.of(context).colorScheme;
+    return Scaffold(
+      appBar: LemonAppBar(
+        title: t.quest.quests,
+      ),
+      backgroundColor: colorScheme.primary,
+      body: SafeArea(
+        child: BlocBuilder<GetPointGroupsBloc, GetPointGroupsState>(
+          builder: (context, state) {
+            final pointGroups = state.pointGroups;
+            final selectedSecondaryLevelGroup = state.selectedSecondLevelGroup;
+            List<String?> pointGroupsTitle =
+                pointGroups.map((item) => item.firstLevelGroup?.title).toList();
+            List<Tab> tabs = pointGroupsTitle.map((tabName) {
+              return Tab(text: StringUtils.capitalize(tabName));
+            }).toList();
+
+            return DefaultTabController(
+              length: tabs.length,
+              initialIndex: 0,
+              child: Column(
+                mainAxisSize: MainAxisSize.max,
+                children: [
+                  TabBar(
+                    onTap: (index) {
+                      setState(() {
+                        selectedTabIndex = index;
+                      });
+                    },
+                    labelStyle: Typo.medium.copyWith(
+                      color: colorScheme.onPrimary,
+                      fontWeight: FontWeight.w500,
+                    ),
+                    unselectedLabelStyle: Typo.medium.copyWith(
+                      color: colorScheme.onSurfaceVariant,
+                      fontWeight: FontWeight.w500,
+                    ),
+                    indicatorColor: LemonColor.paleViolet,
+                    tabs: tabs,
+                  ),
+                  Expanded(
+                    child: TabBarView(
+                      children: pointGroups.map((pointGroup) {
+                        final secondaryLevelGroups =
+                            pointGroup.secondLevelGroups;
+                        return Column(
```

Comment:
I guess this is a widget for each tab screen , let make it a widget instead of a render function